src/datasources/cache/cache.router.ts

```diff
@@ -0,0 +1,265 @@
+import { CacheDir } from './entities/cache-dir.entity';
+
+export class CacheRouter {
```

Comment:
My biggest concern with the `CacheRouter` is that we introduce a component that is now aware of all the features on what we consider to be the caching layer.

Previously this data was provided by each feature, and on a feature basis. I'll try to think of an alternative solution but I do agree that we need a mechanism to easily retrieve/delete/update these entries.

Reply:
I agree, this could be more polished by decoupling the data to create the keys from this class. Thank you for your thoughts on this. I'd go with the current solution to speed up the delivery as you said, but we could decouple this in the future by another mechanism (maybe a register as we have for AJV schemas? That way each feature could register its `CacheDirs` on the application startup).